### PR TITLE
google free and metrics free

### LINKED ISSUE #13293: Remove android remote push notifications
We should remove android remote notifications and any google dependency so that play store & fdroid are the same build (minus signing). 
We need to try to see if referrals still works on that build.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #13293

removed referrals
removed remote notifications
removed metrics

image

### CODE REVIEW

**Comment by Parveshdhull** [comment]
Hi @flexsurfer, Please also remove plugin from [android/build.gradle](https://github.com/status-im/status-react/blob/c3e599721f4d9e7160469c0cd0d64e3bb51d41c8/android/build.gradle#L25) file

**Review by Parveshdhull — APPROVED** [review]

File: nix/deps/gradle/deps.list [hunk 1]
```diff
@@ -396,27 +375,6 @@ com.google.errorprone:error_prone_annotations:2.0.18
...
-com.google.firebase:firebase-iid:21.1.0
-com.google.firebase:firebase-installations-interop:16.0.1
-com.google.firebase:firebase-installations-interop:17.0.0
-com.google.firebase:firebase-installations:16.3.5
-com.google.firebase:firebase-installations:17.0.0
-com.google.firebase:firebase-measurement-connector:18.0.0
-com.google.firebase:firebase-measurement-connector:19.0.0
-com.google.firebase:firebase-messaging:21.1.0
-com.google.firebase:firebase-messaging:23.0.0
 com.google.gms:google-services:4.3.10
```
Comment (jakubgs) [inline nix/deps/gradle/deps.list]: Why does this still remain though?

**Review by jakubgs — APPROVED** [review]
You must have enjoyed this one.

image

File: nix/deps/gradle/deps.list (hunk 1 above)
Comment (flexsurfer) [reply nix/deps/gradle/deps.list]: mhm, good catch

Comment (flexsurfer) [reply nix/deps/gradle/deps.list]: thanks , fixed

**Comment by flexsurfer** [comment]
@jakubgs again :( build failed after i generated deps

**Comment by jakubgs** [comment]
Will take a look in a bit.

**Comment by jakubgs** [comment]
Oh yeah, looks like it's the same thing like last time, it just missed two JARs, probably due to rate limiting or network issues:
```diff
diff --git a/nix/deps/gradle/deps.json b/nix/deps/gradle/deps.json
index 2093a1c9a..fa5829d38 100644
--- a/nix/deps/gradle/deps.json
+++ b/nix/deps/gradle/deps.json
@@ -244,6 +244,10 @@
     "pom": {
       "sha1": "f99012a3ab10fd604a8631aaa8d525aa44df5da5",
       "sha256": "08226chqfsadvgk08yg4bnf67hmqdvl4cvgdhm76n56xkm2isvsb"
+    },
+    "jar": {
+      "sha1": "bb21b9a11761451b51624ac428d1f1bb5deeac38",
+      "sha256": "1n3sqpd398a23jdqwpql9ls0if23vzhaj37fn1j6wllvfwvv702b"
     }
   },
 
@@ -268,6 +272,10 @@
     "pom": {
       "sha1": "2b3b788454b0aa77266f2ce191eba3a7768678ca",
       "sha256": "00fgwnngz2qbicwajshx599l3r0ljjslrjb6qc561g5a1sbb7fw3"
+    },
+    "jar": {
+      "sha1": "b3152fc64428c9354344bd89848ecddc09b6f07e",
+      "sha256": "129qclk47ifk57y8d135lfqchx7gffpswfgy55zkw1lx0azkf4py"
     }
   },
```

**Comment by jakubgs** [comment]
The problem is that there are some some valid cases where only a POM is available, for example:
```json
  {
    "path": "com/google/auto/auto-parent/3/auto-parent-3",
    "host": "https://repo.maven.apache.org/maven2",
    "type": "pom",
    "pom": {
      "sha1": "8e22bd67b5aeb5f03fd50efd5910ba6dfb7f4c86",
      "sha256": "1a78xc2zm7ixc1499amka8i6kq0cqm0f123kq894glngg04ip3y9"
    }
  },
```
But in theory, we could prevent this by checking the `packaging` field in the XML, which indicates whether the package should have a JAR attached to it or not:
```
 > c https://repo.maven.apache.org/maven2/com/google/auto/auto-parent/3/auto-parent-3.pom | grep packaging 
  <packaging>pom</packaging>

 > c https://repo.maven.apache.org/maven2/com/google/code/gson/gson/2.2.4/gson-2.2.4.pom | grep packaging
  <packaging>jar</packaging>
```

**Comment by jakubgs** [comment]
I can make the `nix/deps/gradle/url2json.sh` script fail when a JAR is missing, but there are 3 JARs that are ALWAYS missing:
```
Regenerating Nix files...
Found 39 sub-projects...
Found 651 direct dependencies...
Found 1042 dependency URLs...
 ! Failed to fetch: https://repo.maven.apache.org/maven2/org/codehaus/woodstox/wstx-asl/4.0.6/wstx-asl-4.0.6.jar
 ! Failed to fetch: https://repo.maven.apache.org/maven2/org/eclipse/sisu/org.eclipse.sisu.inject/0.9.0.M1/org.eclipse.sisu.inject-0.9.0.M1.eclipse-plugin
 ! Failed to fetch: https://repo.maven.apache.org/maven2/org/eclipse/sisu/org.eclipse.sisu.plexus/0.9.0.M1/org.eclipse.sisu.plexus-0.9.0.M1.eclipse-plugin
make: *** [Makefile:94: nix-update-gradle] Error 3
```
So that's an issue. But tat can be mitigated based on the contents of their POMs.

**Comment by jakubgs** [comment]
I improved the Nix Gradle update process by failing explicitly on missing JARs:

* https://github.com/status-im/status-react/pull/13369

**Comment by qoqobolo** [comment]
Thanks @flexsurfer
 
Tested:
- three types of referral links
- local notifications on Android devices both with and without Google services
- upgrade from the Play Store build (1.19) to the manual build created from this branch

Works fine, e2e failures are not related to PR, it can be merged.

Screenshot 2022-05-18 at 17 03 27

**Pull request merged**